src/charm.py

```diff
@@ -69,10 +69,11 @@ def _update_prometheus_jobs(self, _):
         if not self.unit.is_leader():
             return
 
-        if jobs := self._scrape_jobs():
-            for relation in self.model.relations[self._prometheus_relation]:
-                relation.data[self.app]["scrape_jobs"] = json.dumps(jobs)
+        jobs = self._scrape_jobs()
+        for relation in self.model.relations[self._prometheus_relation]:
+            relation.data[self.app]["scrape_jobs"] = json.dumps(jobs)
 
+        if jobs:
             self.unit.status = ActiveStatus()
```

Comment:
Should there be an `else` here? Otherwise after targets is blanked, status would remain as Active.

Reply:
Nop, after targets is blanked, status is `Blocked` (see line 101)